### PR TITLE
bug fix: heatmap issues, custom variables

### PR DESCRIPTION
**Pull request by swaathee**
Fixes the following issues 
1. Heatmap does not work for elements inside shadow dom
2. Scroll map height is not set properly 
3. Scrollmap canvas crashes for pages with a very large height 
4.  Custom variables are overriden in some cases and appended in others
5. Add Json+LD for article headlines and creators

### CODE REVIEW

**Review by toby-walker — APPROVED** [review]
:shipit:

File: packages/clarity-js/src/layout/node.ts [hunk 1]
```diff
@@ -104,6 +105,20 @@ export default function (node: Node, source: Source): Node {
                     break;
                 case "NOSCRIPT":
                 case "META":
```
Comment (sarveshnagpal) [inline packages/clarity-js/src/layout/node.ts:107]: Let's talk offline about this - since this is a lot of extra code we just introduced.

**Review by sarveshnagpal — CHANGES_REQUESTED** [review]
Let's talk about OG offline. Thanks.

File: packages/clarity-js/src/layout/node.ts (hunk 1 above)
Comment (sarveshnagpal) [reply packages/clarity-js/src/layout/node.ts:107]: Synced offline - will revisit this in future iterations.

File: packages/clarity-js/src/layout/node.ts [hunk 2]
```diff
@@ -104,6 +105,20 @@ export default function (node: Node, source: Source): Node {
...
+                    if (Constant.Property in attributes && Constant.Content in attributes) {
+                        var content = attributes[Constant.Content]
```
Comment (sarveshnagpal) [inline packages/clarity-js/src/layout/node.ts:109]: We don't use var anymore. Use let for now. We will make a pass and clean everything else.

File: packages/clarity-js/src/layout/node.ts [hunk 3]
```diff
@@ -104,6 +105,20 @@ export default function (node: Node, source: Source): Node {
                     break;
                 case "NOSCRIPT":
```
Comment (sarveshnagpal) [inline packages/clarity-js/src/layout/node.ts:106]: Add break here so we don't go through NOSCRIPT tag

**Review by sarveshnagpal — APPROVED** [review]

**Pull request merged**